POLIO-750 Preparedness: send notifications when data is outside range (0-100)

olethanh opened the pull request:
Added to existing cron sending alert if value are out of range
To control who receive the e-mail create edit a polio.Config object with slug emails_for_preparedness_alert
  and containing a list of e-mail as json.
  e.g. ['test@bluesquarehub.com']

Related JIRA tickets : POLIO-750


## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [NA] Are my typescript files well typed
- [NA] Did I add translations
- [NA] My migrations file are included
- [NA] Are there enough tests


## Notes

in the end Since it's only once per day and I filtered only on up coming round and I send one e-mail for all the campaign, I didn't include a mechanism to not double send notification, it made the logic a lot simpler (otherwhise we could miss regression).

this is build on top of the caching branch  POLIO-725

quang-le commented:
## Update 20/08/2021

- add styling to cursor

olethanh commented:
![image](https://user-images.githubusercontent.com/82500/130415345-f6bef64a-6963-48fb-ba40-61878bd41b97.png)
j'ai l'impression l'espace entre les colonnes est un peu trop grand, ca revient a la ligne alors qu'il reste plein d'espace

quang-le commented on plugins/polio/js/src/components/MapComponent/MapComponent.js:
```diff
@@ -42,17 +53,42 @@ export const MapComponent = ({ onSelectShape, shapes, getShapeStyle }) => {
-                        <Tooltip>{shape.name}</Tooltip>
-                    </GeoJSON>
-                ))}
+            <Pane name="Regions">
```
> ![image](https://user-images.githubusercontent.com/82500/130415345-f6bef64a-6963-48fb-ba40-61878bd41b97.png)
> I feel like the space between the column is a bit to big, it goes to the line when there is plenty of space left
If you refer to "CROSS RIVER" in the video, the name is actually too long for the column width, but the table wraps it at the space

olethanh commented:
![image](https://user-images.githubusercontent.com/82500/130422899-ecdcd8af-e2da-4a28-82d0-4836e2453dab.png)
I added an image but it didn't work, for example here, North Horr shouldn't warp.

The actions seems to be center aligned, maybe we should also align the title?

quang-le commented:
> ![image](https://user-images.githubusercontent.com/82500/130422899-ecdcd8af-e2da-4a28-82d0-4836e2453dab.png)
> I added an image but it didn't work, for example here, North Horr shouldn't warp.
> 
> The actions seems to be center aligned, maybe we should also align the title?
> 
> maybe we could also decrease the font size.

Oh, I didn't realize the icon button's padding screwed the alignment. Nice catch. I'll make the font smaller as well

olethanh committed:
improve code style

olethanh committed:
add prefrech to reduce queries

olethanh committed:
Add cache on preparedness per round

olethanh committed:
reorganize preparedness caching

olethanh committed:
update cache when we save the campaign too

olethanh committed:
Reorganize code, recalculate when we save campaign, adapttests

olethanh committed:
POLIO-750 Preparedness: send notifications when data is outside range (0-100)

To control who receive the e-mail create edit a polio.Config object with slug emails_for_preparedness_alert
and containing a list of e-mail as json.
e.g. ['test@bluesquarehub.com']

olethanh committed:
fix mypy
